orbit/pkg/table/software_update/software_update.go

```diff
@@ -0,0 +1,58 @@
+//go:build darwin
+// +build darwin
+
+package software_update
+
+import (
+	"context"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/osquery/osquery-go/plugin/table"
+	"github.com/rs/zerolog/log"
+)
+
+// Columns is the schema of the table.
+func Columns() []table.ColumnDefinition {
+	return []table.ColumnDefinition{
+		table.IntegerColumn("software_update_required"),
+	}
+}
+
+// Generate is called to return the results for the table at query time.
+// Constraints for generating can be retrieved from the queryContext.
+func Generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	newSoftwareAvailable, err := isNewSoftwareAvailable(ctx)
+
+	return []map[string]string{
+		{"software_update_required": newSoftwareAvailable},
+	}, err
+}
+
+func isNewSoftwareAvailable(ctx context.Context) (newSoftwareAvailable string, err error) {
+	res, err := runCommand(ctx, "/usr/sbin/softwareupdate", "-l")
+	newSoftwareAvailable = ""
+	if err == nil {
+		newSoftwareAvailable = "1"
+		if strings.Contains(res, "No new software available") {
+			newSoftwareAvailable = "0"
+		}
+	}
+	return newSoftwareAvailable, err
+}
+
+func runCommand(ctx context.Context, name string, arg ...string) (res string, err error) {
+	// This query may take more than the avg query
+	// I doubled the typical time from my tests and ended up with 30 seconds timeout.
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+	cmd := exec.CommandContext(ctx, name, arg...)
+
+	out, err := cmd.Output()
```

Comment:
(This is not working on my up-to-date workstation.) It seems you must use `CombinedOutput`:
```
$ /usr/sbin/softwareupdate -l 2>err.txt 1>out.txt

$ cat out.txt
Software Update Tool

Finding available software

$ cat err.txt
No new software available.
```